tests/system/Validation/ValidationTest.php

```diff
@@ -87,6 +88,59 @@ public function testSetRulesStoresRules()
         $this->assertEquals($rules, $this->validation->getRules());
     }
 
+    public function testSetRuleStoresRule()
+    {
+        $this->validation->setRules([]);
+        $this->validation->setRule('foo', null, 'bar|baz');
+
+        $this->assertEquals([
+            'foo' => [
+                'label' => null,
+                'rules' => 'bar|baz',
+            ],
+        ], $this->validation->getRules());
+    }
+
+    public function testSetRuleAddsRule()
+    {
+        $this->validation->setRules([
+            'bar' => [
+                'label' => null,
+                'rules' => 'bar|baz',
+            ],
+        ]);
+        $this->validation->setRule('foo', null, 'foo|foz');
+
+        $this->assertEquals([
```

Comment:
assertSame

Reply:
assertSame can fail when the order of fields is not met. A changed order doesn't bother us, so assertEquals would do a better job there, or am I wrong?

Reply:
It is true that production-wise ordering is not important. But our tests now shifts to stricter assertions to catch lousy errors.

In your case the order can be seen easily as in your array union the new rule comes first followed by existing rules so you can easily come up with the correct order.